pkg/reconciler/instances/scmigration/sc_removal.go

```diff
@@ -0,0 +1,375 @@
+			Kind:    "ServiceBroker",
+			Group:   "servicecatalog.k8s.io",
+			Version: "v1beta1",
+		},
+		{
+			Kind:    "AddonsConfiguration",
+			Group:   "addons.kyma-project.io",
+			Version: "v1alpha1",
+		},
+	}
+
+	namespaces := &v1.NamespaceList{}
+	err := c.k8sCli.List(ac.Context, namespaces)
+	if err != nil {
+		return err
+	}
+
+	var errs []error
+	for _, gvk := range gvkList {
+		for _, namespace := range namespaces.Items {
+			ac.Logger.Infof("deleting %s in %s\n", gvk.Kind, namespace.Name)
+			u := &unstructured.Unstructured{}
+			u.SetGroupVersionKind(gvk)
+			err = c.k8sCli.DeleteAllOf(ac.Context, u, client.InNamespace(namespace.Name))
+			if meta.IsNoMatchError(err) {
+				ac.Logger.Infof("CRD for GVK %s not found, skipping resource deletion", gvk)
+				break
+			} else if err != nil {
+				errs = append(errs, gerr.Wrap(err, fmt.Sprintf("deleting all resources %v in %v", gvk, namespace.Name)))
+			}
+		}
+	}
+
+	clusterGVKList := []schema.GroupVersionKind{
+		{
+			Kind:    "ClusterAddonsConfiguration",
+			Group:   "addons.kyma-project.io",
+			Version: "v1alpha1",
+		},
+		{
+			Kind:    "ClusterServiceBroker",
+			Group:   "servicecatalog.k8s.io",
+			Version: "v1beta1",
+		},
+	}
+	for _, gvk := range clusterGVKList {
+		u := &unstructured.Unstructured{}
+		u.SetGroupVersionKind(gvk)
+		err = c.k8sCli.DeleteAllOf(ac.Context, u, client.InNamespace(""))
+		if meta.IsNoMatchError(err) {
+			ac.Logger.Infof("CRD for GVK %s not found, skipping resource deletion", gvk)
+			continue
+		} else if err != nil {
+			errs = append(errs, gerr.Wrap(err, fmt.Sprintf("deleting all resources %v", gvk)))
+		}
+	}
+	return errors.NewAggregate(errs)
+}
+
+func (c *scremoval) waitForPodsGone(ac *service.ActionContext, dep unstructured.Unstructured) error {
+	path := []string{"spec", "selector", "matchLabels"}
+	ls, found, err := unstructured.NestedStringMap(dep.Object, path...)
+	if err != nil {
+		msg := fmt.Sprintf("unstructured dep %v/%v failed to find selector %v: %v", dep.GetNamespace(), dep.GetName(), path, err)
+		return gerr.Wrap(err, msg)
+	}
+	if !found {
+		return fmt.Errorf("unstructured dep %v/%v missing selector %v", dep.GetNamespace(), dep.GetName(), path)
+	}
+	cnd := func() (bool, error) {
+		pods := &v1.PodList{}
+		opts := []client.ListOption{client.InNamespace(dep.GetNamespace()), client.MatchingLabels(ls)}
+		if err := c.k8sCli.List(ac.Context, pods, opts...); err != nil {
+			return false, err
+		}
+		return len(pods.Items) == 0, nil
+	}
+	return wait.PollImmediate(5*time.Second, 10*time.Minute, cnd)
+}
+
+func (c *scremoval) ensureServiceCatalogNotRunning(ac *service.ActionContext) error {
+	done := make(chan error, 2)
+	go func() {
+		done <- c.waitForPodsGone(ac, serviceCatalogCatalogControllerManagerUnstructuredDeployment)
+	}()
+	go func() {
+		done <- c.waitForPodsGone(ac, helmBrokerUnstructuredDeployment)
+	}()
+	var errs []error
+	errs = append(errs, <-done)
+	errs = append(errs, <-done)
+	return errors.NewAggregate(errs)
```

Comment:
fwiw, the `NewAggregate` filters out `nil` errors, if the slice then is empty, it returns `nil` as well 

https://github.com/kubernetes/apimachinery/blob/df993592a122931b8aac4db57689e09458a2332c/pkg/util/errors/errors.go#L52-L59